### PR TITLE
Support deprecated tags

### PR DESCRIPTION
**Pull request by mglaman**
It seems like this would make handling multiline deprecated messages easier, like Throws or Return tags https://github.com/phpstan/phpstan/pull/1983

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Hi, this looks okay but a test would be nice :) We can then update 1983.

**Comment by mglaman** [comment]
Yes, I will do that. Had to run before I could add, but wanted to get something up before I forgot 😄

**Comment by mglaman** [comment]
I have added some tests. I wonder if this will help handle deprecations in PHPStan and the messaging. Hopefully :)

**Comment by JanTvrdik** [comment]
I don't understand what is this PR for. Parsing `@deprecated` tags is already supported. I does not need any special code.

**Comment by mglaman** [comment]
Technically they can be discovered. Other tags have typed objects. I was hoping this would improve parsing of deprecated messages in PHPStan itself per my PR linked previously

**Comment by ondrejmirtes** [comment]
I think that if there are other types of nodes in the switch then there should be `@deprecated` as well - simplifies the usage for phpdoc-parser users.

**Comment by ondrejmirtes** [comment]
Please fix the build. Now all jobs fail. (Failure on 7.3 is fine, I need to update PHPCS on master.)

**Comment by JanTvrdik** [comment]
> Other tags have typed objects

It already has a typed object (`GenericTagValueNode`).

> I was hoping this would improve parsing of deprecated messages 

It uses the same parsing the default branch.

File: src/Parser/PhpDocParser.php [hunk 1]
```diff
@@ -159,6 +163,12 @@ private function parseThrowsTagValue(TokenIterator $tokens): Ast\PhpDoc\ThrowsTa
 		return new Ast\PhpDoc\ThrowsTagValueNode($type, $description);
 	}
 
+	private function parseDeprecatedTagValue(TokenIterator $tokens): Ast\PhpDoc\DeprecatedTagValueNode
+	{
+		$description = $this->parseOptionalDescription($tokens, true);
```
Comment (JanTvrdik) [inline src/Parser/PhpDocParser.php]: the second parameter should be `false`

**Comment by ondrejmirtes** [comment]
`DeprecatedTagValueNode` is a little bit more type-safe but I understand it's a small gain. Anyway, I wouldn't want to block custom parsing for this because other planned tags like `@template` will need it too.

**Comment by mglaman** [comment]
@ondrejmirtes I was finally able to circle back to this now that DrupalCon is over. I have PHP 7.3 on all of my locals, so took a little bit to fix PHPCS but it's all looking green again.

**Comment by mglaman** [comment]
@ondrejmirtes 🙌 thanks!

**Comment by ondrejmirtes** [comment]
I rebased, made some changes and merged it :) For example the description shouldn't have been nullable, it was never null besides the test case.

**Comment by mglaman** [comment]
Thanks! Appreciate that, helping me learn a few things.

**Pull request merged**